### DISCUSSION TITLE
Is it possible to separate task types

### DISCUSSION THREAD

**Discussion by ghost** [Leantime/leantime#554] [Q&A] [answered]
Basically, how to differentiate dev from architecture from testing tasks ? I see we can add tags, though it doesnt looks like there is any use of them. Is there is subtype that could be use to color tasks by type, or to see the overall time spent on each task type ?

**Comment by marcelfolaron** [comment]
We do have task types already and adding colors should not be an issue. I would probably do it the way it is on the dashboard and add a colored border to the left side of the card.

**Reply by ghost** [reply]
Yes but i mean you cant add new tasks types without editing the source and rebuilding

**Reply by ghost** [reply]
```php
        public $type = array('integration','dev','3D','GD','research','testing','bug','other' );
        public $typeIcons = array('integration'=>'fa-unity','dev'=>'fa-code','3D'=>'fa-cube','GD'=>'fa-chalkboard','research'=>'fa-search','testing'=>'fa-flask','bug'=> 'fa-bug','other'=>'fa-check-square');
```
I've added my new types, but for them to display properly i need to add some entry in the language files. And i think i need to re-build the project for them to work ?
![image](https://user-images.githubusercontent.com/2307228/148241927-07c3eb6f-95cd-40f8-9635-86d4725f01bd.png)

**Comment by marcelfolaron** [comment]
No need to rebuild. (There is not really a build process since it's PHP).
The types are cached. So if you log out and back in, they should show up.

**Reply by ghost** [reply]
Humm.. Weird because i did close everything, cleared the browser cache, disconnect, reconnect, and still the labels arent updated.

**Comment by marcelfolaron** [comment, marked as answer]
Sorry, yes you do need to add the translation strings into your language
file. I suggest you copy the language file to something like en-Custom.ini
and then add those strings.
Which screen do you use to verify the types are included? I looked at
showTicket and it should just take the values from the type array.

**Reply by ghost** [reply]
Ok so i did add the translation strings but for whatever reason, it wouldnt work until i add them to the en-US as well, even though i'm using en-GB
I was indeed using showticket
Btw i noticed that the "filter" in todo lists, doesnt use the translated value but the key directly. I wanted to make a pull request, but i cant get composer to work on my system so i'll just leave it as is.
Thank you for your help :)
